lighthouse-core/audits/seo/tap-targets.js

```diff
@@ -0,0 +1,334 @@
+      if (failure) {
+        // only update our state if this was the biggest failure we've seen for this pair
+        if (!greatestFailure ||
+          failure.overlapScoreRatio > greatestFailure.overlapScoreRatio) {
+          greatestFailure = {
+            ...failure,
+            tapTarget,
+            overlappingTarget: maybeOverlappingTarget,
+          };
+        }
+      }
+    }
+  }
+  return greatestFailure;
+}
+
+/**
+ * @param {LH.Artifacts.Rect} targetCR
+ * @param {LH.Artifacts.Rect} maybeOverlappingCR
+ * @returns {ClientRectOverlapFailure | null}
+ */
+function getOverlapFailureForClientRectPair(targetCR, maybeOverlappingCR) {
+  const fingerRect = getRectAtCenter(targetCR, FINGER_SIZE_PX);
+  // Score indicates how much of the finger area overlaps each target when the user
+  // taps on the center of targetCR
+  const tapTargetScore = getRectOverlapArea(fingerRect, targetCR);
+  const maybeOverlappingScore = getRectOverlapArea(fingerRect, maybeOverlappingCR);
+
+  const overlapScoreRatio = maybeOverlappingScore / tapTargetScore;
+  if (overlapScoreRatio < MAX_ACCEPTABLE_OVERLAP_SCORE_RATIO) {
+    // low score means it's clear that the user tried to tap on the targetCR,
+    // rather than the other tap target client rect
+    return null;
+  }
+
+  return {
+    overlapScoreRatio,
+    tapTargetScore,
+    overlappingTargetScore: maybeOverlappingScore,
+  };
+}
+
+/**
+ * Only report one failure if two targets overlap each other
+ * @param {TapTargetOverlapFailure[]} overlapFailures
+ * @returns {TapTargetOverlapFailure[]}
+ */
+function mergeSymmetricFailures(overlapFailures) {
+  /** @type TapTargetOverlapFailure[] */
+  const failuresAfterMerging = [];
+
+  overlapFailures.forEach((failure, overlapFailureIndex) => {
+    const symmetricFailure = overlapFailures.find(f =>
+      f.tapTarget === failure.overlappingTarget &&
+      f.overlappingTarget === failure.tapTarget
+    );
+
+    if (!symmetricFailure) {
+      failuresAfterMerging.push(failure);
+      return;
+    }
+
+    const {overlapScoreRatio: failureOSR} = failure;
+    const {overlapScoreRatio: symmetricOSR} = symmetricFailure;
+    // Push if:
+    // - the current failure has a higher OSR
+    // - OSRs are the same, and the current failure comes before its symmetric partner in the list
+    // Otherwise do nothing and let the symmetric partner be pushed later.
+    if (failureOSR > symmetricOSR || (
+      failureOSR === symmetricOSR &&
+      overlapFailureIndex < overlapFailures.indexOf(symmetricFailure)
+    )) {
+      failuresAfterMerging.push(failure);
+    }
+  });
+
+  return failuresAfterMerging;
+}
+
+/**
+ * @param {TapTargetOverlapFailure[]} overlapFailures
+ * @returns {LH.Audit.TapTargetTableItem[]}
+ */
+function getTableItems(overlapFailures) {
+  const tableItems = overlapFailures.map(
+    ({
+      tapTarget,
+      overlappingTarget,
+      tapTargetScore,
+      overlappingTargetScore,
+      overlapScoreRatio,
+    }) => {
```

Comment:
style nit: I don't personally find this very readable, not sure how others feel about it. You've lost track of the `map()` by the time you get to the bottom of the destructured props :) I'd rather just have something like
```js
const tableItems = overlapFailures.map(failure => {
  const largestCR = getLargestRect(failure.tapTarget.clientRects);
  // ...
});
```